inquire/src/prompts/path_select/prompt.rs

```diff
@@ -0,0 +1,567 @@
+use crate::{
+    error::InquireResult,
+    formatter::MultiOptionFormatter,
+    input::{Input, InputActionResult},
+    list_option::ListOption,
+    prompts::{
+        path_select::{
+            PathEntry, PathSelect, PathSelectConfig, PathSelectPromptAction, PathSelectionMode,
+        },
+        prompt::{ActionResult, Prompt},
+    },
+    ui::MultiSelectBackend,
+    utils::paginate,
+    validator::ErrorMessage,
+    InquireError,
+};
+use std::{
+    collections::{BTreeSet, HashSet},
+    convert::TryFrom,
+    env,
+    path::{Component, Path, PathBuf},
+};
+
+pub struct PathSelectPrompt<'a> {
+    message: &'a str,
+    options: Vec<PathEntry>,
+    #[allow(dead_code)]
+    divider: &'a str,
+    show_symlinks: bool,
+    select_multiple: bool,
+    filtered_options: Vec<usize>,
+    data_needs_refresh: bool,
+    help_message: Option<&'a str>,
+    config: PathSelectConfig,
+    show_hidden: bool,
+    cursor_index: usize,
+    #[allow(dead_code)]
+    divider_index: usize,
+    selected: HashSet<PathEntry>,
+    checked: BTreeSet<usize>,
+    input: Input,
+    formatter: MultiOptionFormatter<'a, PathEntry>,
+    error: Option<ErrorMessage>,
+    current_path: PathBuf,
+    selection_mode: PathSelectionMode<'a>,
+}
+
+impl<'a> PathSelectPrompt<'a> {
+    pub fn new<T: AsRef<Path>>(pso: PathSelect<'a, T>) -> InquireResult<Self> {
+        let config = PathSelectConfig::from(&pso);
+
+        if let Some(default) = pso.default {
+            default.iter().try_for_each(|default_item| {
+                // Are all of the selected files extant?
+                let default_path = default_item.as_ref();
+                if !default_path.exists() {
+                    Err(InquireError::InvalidConfiguration(format!(
+                        "Specified default path `{default_path:?}` does not exist"
+                    )))
+                } else {
+                    Ok(())
+                }
+            })?;
+        }
```

Comment:
I think the behavior for default will be better if it's "selected if exists", without raising an error if not.

The why is that, thinking from the perspective of the developer using inquire, they're gonna have to make a very strong assumption on this path existing on the _user's machine_, and if they don't exist by any reason, this will be an unrecoverable error for the user.

This will force everyone using that feature to check if the path exists before creating this default list, which will be extra boilerplate we can remove if the goal is "if this path exists, select it".

In other words, if a path existing is so important for me that the prompt becomes irrelevant (should error out) if it doesn't exist, I as a developer likely want to make that assertion myself, with my own error handling, rather than delegating this check to inquire.

I'm just going through the code but I still want to write a few examples and see how the prompt feels UX-wise before making any assertions, so once I actually use this prompt (as you probably are already), I might change my idea here if I'm speaking non-sense.

Reply:
That makes sense. I still ended up filtering out non-existent supplied paths because it seemed to be the easiest way to avoid errors that would come up in rendering.